{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "07d7dee6-e81b-465e-89f1-b3f46efda4b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Make_tree2array(names_file):\n",
    "    names = {}\n",
    "\n",
    "    for line in names_file:\n",
    "        if line == \"\\n\":\n",
    "            #print(\"empty row\")\n",
    "            continue\n",
    "\n",
    "        data = [attr.strip() for attr in line.split(\"|\")[0:2]]\n",
    "        names[data[1]] = int(data[0])\n",
    "\n",
    "    return names\n",
    "\n",
    "\n",
    "def Nodes2Array(nodes_file):\n",
    "    nodes = {}\n",
    "\n",
    "    for line in nodes_file:\n",
    "        if line == \"\\n\":\n",
    "            #print(\"empty row\")\n",
    "            continue\n",
    "\n",
    "        data = [attr.strip() for attr in line.split(\"|\")[0:3]]\n",
    "\n",
    "        nodes[int(data[0])] = [int(data[1]), data[2]]\n",
    "    return nodes\n",
    "\n",
    "\n",
    "def Find_Value2Key_in_Names(Key, nodes, names, Taxon):\n",
    "    if Key in sorted(names):\n",
    "        Key_in_nodes = names[Key]\n",
    "        if Key_in_nodes in sorted(nodes):\n",
    "            if nodes[Key_in_nodes][1] == Taxon:\n",
    "                #print(\"FOUND THE KEY\")\n",
    "                return Key\n",
    "            # ----------------------------------------------------------------------------\n",
    "            else:\n",
    "                Find_Value2Key_in_Names(nodes[Key_in_nodes][0], nodes, names, Taxon)\n",
    "        else:\n",
    "            print(\"Key_in_nodes cannot be found.\")\n",
    "    else:\n",
    "        #print(\"Key cannot be found.LABEL\")\n",
    "        return \"UNKNOWN\"\n",
    "    #print(\"\")\n",
    "\n",
    "\n",
    "\n",
    "names_file = open(\"csado7/SILVA132/taxonomy/names.dmp\", \"r\")\n",
    "nodes_file = open(\"csado7/SILVA132/taxonomy/nodes.dmp\", \"r\")\n",
    "\n",
    "names = Make_tree2array(names_file)\n",
    "nodes = Nodes2Array(nodes_file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "79a7f1d8-6d1d-4ae7-9ca5-a3ebc4e90898",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.9382497433928556\n",
      "5.876499486785711\n",
      "8.814749230178567\n",
      "11.752998973571422\n",
      "14.691248716964278\n",
      "17.629498460357134\n",
      "20.567748203749993\n",
      "23.505997947142845\n",
      "26.4442476905357\n",
      "29.382497433928556\n",
      "32.320747177321415\n",
      "35.25899692071427\n",
      "38.197246664107126\n",
      "41.135496407499986\n",
      "44.07374615089283\n",
      "47.01199589428569\n",
      "49.95024563767855\n",
      "52.8884953810714\n",
      "55.82674512446426\n",
      "58.76499486785711\n",
      "61.70324461124997\n",
      "64.64149435464283\n",
      "67.57974409803568\n",
      "70.51799384142853\n",
      "73.4562435848214\n",
      "76.39449332821425\n",
      "79.3327430716071\n",
      "82.27099281499997\n",
      "431575\n",
      "3269\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "import operator\n",
    "import math\n",
    "\n",
    "import Bio\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "seqs = []\n",
    "groups = []\n",
    "\n",
    "transformed_group = []\n",
    "\n",
    "class GroupedByGenus_reads():\n",
    "    \n",
    "    def __init__(self,Genus,ListOfGenus):\n",
    "        self.Genus = Genus\n",
    "        self.ListOfGenus = ListOfGenus\n",
    "        self.reads = []\n",
    "     \n",
    "    def _add(self,element):\n",
    "        self.ListOfGenus.append(element)\n",
    "    \n",
    "    def _read_list_extend(self, element):\n",
    "        self.reads+=element\n",
    "    \n",
    "            \n",
    "    def _add(self,element):\n",
    "        self.ListOfGenus.append(element)\n",
    "\n",
    "    def fill_reads(self,isFasta=True):\n",
    "        asd=0\n",
    "        for temp in self.ListOfGenus:\n",
    "            #self.reads +=self.create_reads(temp.seq,temp.name)\n",
    "            asd+=1\n",
    "            if isFasta:\n",
    "                self.reads.extend(create_reads_fasta(temp.seq,temp.name,self.Genus))\n",
    "            else:\n",
    "                self.reads.extend(create_reads(temp.seq,temp.name,self.Genus))\n",
    "            \n",
    "            \n",
    "            \n",
    "            if asd%5000==0:\n",
    "                print(str(asd))\n",
    "                 \n",
    "\n",
    "class GroupedByGenus():\n",
    "    \n",
    "    def __init__(self,Genus,ListOfGenus):\n",
    "        self.Genus = Genus\n",
    "        self.ListOfGenus = ListOfGenus\n",
    "        \n",
    "    def _add(self,element):\n",
    "        self.ListOfGenus.append(element)\n",
    "\n",
    "count=int(0)\n",
    "with open(\"csado7/SILVA132/SILVA/library/silva.fna\") as input_handle:\n",
    "    for record in SeqIO.parse(input_handle, \"fasta\"):\n",
    "        \n",
    "        \n",
    "        \n",
    "        #------------------------csak: Bacteria Ã©s 1300<= x hossz <= 1650----------------------------------\n",
    "        \n",
    "        if record.description.split(' ',1)[1].split(';')[0] != \"Bacteria\":\n",
    "            continue\n",
    "        \n",
    "        if len(record.seq)<1300 and len(record.seq)>1650:\n",
    "            continue\n",
    "        \n",
    "        #-------------------------------------------------------------------------\n",
    "        count+=1\n",
    "        \n",
    "        #curr_genus = record.description.split(' ',1)[1].split(';')[-2]\n",
    "        #curr_genus = record.description.split(' ',1)[1].split(';')[6]\n",
    "        curr_genus = str(Find_Value2Key_in_Names(record.description.split(' ',1)[1].split(';')[-2], nodes, names, \"genus\"))\n",
    "        #print(record.description.split(' ',1)[1].split(';')[-1])\n",
    "        \n",
    "        if type(curr_genus) != str:\n",
    "            print(count)\n",
    "            print(\"Ennyi\")\n",
    "            print(record.description)\n",
    "            print(curr_genus)\n",
    "            break\n",
    "   \n",
    "        change =0\n",
    "            \n",
    "        if count %15000==0:\n",
    "            print(int(count)/510508*100)\n",
    "\n",
    "        for i in range(0,len(transformed_group)):\n",
    "            if transformed_group[i].Genus == curr_genus:\n",
    "                #transformed_group[i].ListOfGenus.append(record)\n",
    "                transformed_group[i]._add(record)\n",
    "                change+=1\n",
    "                \n",
    "        if change ==0:\n",
    "            gp = GroupedByGenus_reads(curr_genus,[])\n",
    "            gp._add(record)\n",
    "            #gp.Genus = curr_genus\n",
    "            #gp.ListOfGenus.append(record)\n",
    "            transformed_group.append(gp)\n",
    "        \n",
    "    \n",
    "        #if not curr_genus in groups:\n",
    "        #   groups.append(curr_genus) \n",
    "        #seqs.append(record)\n",
    "        \n",
    "print(count)\n",
    "print(str(len(transformed_group)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "843bd942-68f1-4df9-b99f-236d87bfc73a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "49af3365-a759-4967-834b-95a4112d5dfd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "431575\n"
     ]
    }
   ],
   "source": [
    "import Bio\n",
    "\n",
    "_asd=0\n",
    "for s in transformed_group:\n",
    "#for s in range(0,1):\n",
    "    s_name=s.Genus\n",
    "    \n",
    "    if \" \" in s_name:\n",
    "        s_name= s_name.replace(' ','_')\n",
    "\n",
    "    if \"-\" in s_name:    \n",
    "        s_name= s_name.replace('-','_')\n",
    "    \n",
    "    path= \"MSA_score/Taxonomy_Genus_Just_Bacteria/\"+s_name+\".fasta\"\n",
    "    File_Genus = open(path, 'w')\n",
    "    \n",
    "    for r in s.ListOfGenus: \n",
    "        File_Genus.write(\">\"+r.description+\"\\n\")\n",
    "        File_Genus.write(str(r.seq)+\"\\n\")\n",
    "        _asd+=1\n",
    "        \n",
    "    File_Genus.close()\n",
    "    \n",
    "    \n",
    "    \n",
    "print(str(_asd))\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1d320ce7-affa-49be-88e6-4365b36e66f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3269\n"
     ]
    }
   ],
   "source": [
    "\n",
    "path= \"Taxonomy_Genus_Just_Bacteria/Genus_group_name_WORKING.txt\"\n",
    "File_Genus = open(path, 'w')\n",
    "\n",
    "k=0\n",
    "for s in transformed_group:\n",
    "    k+=1\n",
    "    s_name=s.Genus\n",
    "    if \" \" in s_name:\n",
    "        s_name= s_name.replace(' ','_')\n",
    "\n",
    "    if \"-\" in s_name:    \n",
    "        s_name= s_name.replace('-','_')\n",
    "    \n",
    "\n",
    "    File_Genus.write(s_name+\"\\n\")\n",
    "\n",
    "        \n",
    "File_Genus.close()\n",
    "    \n",
    "    \n",
    "    \n",
    "print(str(k))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4bc06577-9677-4cef-97a6-c45206de4b39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3269\n"
     ]
    }
   ],
   "source": [
    "\n",
    "def create_reads(curr_seq,name,genus,a=30):\n",
    "        return_string= []\n",
    "        coverage = [0] * len(curr_seq)\n",
    "        #--------------------\n",
    "\n",
    "        c = int((len(curr_seq)+50)/(((len(curr_seq)-200)*a)/150))\n",
    "        \n",
    "\n",
    "        #--------------------\n",
    "        x = len(curr_seq)\n",
    "\n",
    "        short_reads = 0\n",
    "\n",
    "        it_end =0\n",
    "\n",
    "        Length_diff =[]\n",
    "\n",
    "        while it_end < len(curr_seq):\n",
    "\n",
    "            if it_end+150 > len(curr_seq):\n",
    "                if it_end + 50 > len(curr_seq):\n",
    "                    it_end = len(curr_seq)+1\n",
    "                    break\n",
    "                else:\n",
    "                    read_end =len(curr_seq)\n",
    "            else:\n",
    "                read_end = it_end+150\n",
    "\n",
    "\n",
    "            for delete in range(it_end,read_end):\n",
    "                coverage[delete] += 1\n",
    "                \n",
    "            \n",
    "            string_out = \"@\"+name + \"|\" +genus+\"|\"+ \"S\"+str(it_end)  + \"E\" + str(read_end) + \"\\n\" \\\n",
    "                         + str(curr_seq[it_end:read_end]) + \"\\n\"+ \"+\" \\\n",
    "            +\"\\n\" +\"H\" * (read_end-it_end)+\"\\n\"\n",
    "            \n",
    "            \n",
    "            return_string.append(string_out)\n",
    "\n",
    "            Length_diff.append(read_end-it_end)\n",
    "\n",
    "\n",
    "            it_end += c\n",
    "        return return_string\n",
    "    \n",
    "def create_reads_fasta(curr_seq,name,genus,a=30):\n",
    "        return_string= []\n",
    "        coverage = [0] * len(curr_seq)\n",
    "        #--------------------\n",
    "\n",
    "        c = int((len(curr_seq)+50)/(((len(curr_seq)-200)*a)/150))\n",
    "        \n",
    "\n",
    "        #--------------------\n",
    "        x = len(curr_seq)\n",
    "\n",
    "        short_reads = 0\n",
    "\n",
    "        it_end =0\n",
    "\n",
    "        Length_diff =[]\n",
    "\n",
    "        while it_end < len(curr_seq):\n",
    "\n",
    "            if it_end+150 > len(curr_seq):\n",
    "                if it_end + 50 > len(curr_seq):\n",
    "                    it_end = len(curr_seq)+1\n",
    "                    break\n",
    "                else:\n",
    "                    read_end =len(curr_seq)\n",
    "            else:\n",
    "                read_end = it_end+150\n",
    "\n",
    "\n",
    "            for delete in range(it_end,read_end):\n",
    "                coverage[delete] += 1\n",
    "                \n",
    "            \n",
    "            string_out = \">\"+name + \"|\" +genus+\"|\"+ \"S\"+str(it_end)  + \"E\" + str(read_end) + \"\\n\" \\\n",
    "                         + str(curr_seq[it_end:read_end]) + \"\\n\"\n",
    "            \n",
    "            \n",
    "            return_string.append(string_out)\n",
    "\n",
    "            Length_diff.append(read_end-it_end)\n",
    "\n",
    "\n",
    "            it_end += c\n",
    "        return return_string\n",
    "\n",
    "    \n",
    "def print_(seq):\n",
    "    print(seq)\n",
    "\n",
    "print(len(transformed_group))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "05bfc34f-5198-491d-9b4a-3d9b42e5e9ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5000\n",
      "5000\n",
      "5000\n",
      "5000\n",
      "10000\n",
      "15000\n",
      "5000\n",
      "5000\n",
      "10000\n",
      "15000\n",
      "20000\n",
      "25000\n",
      "30000\n",
      "35000\n",
      "40000\n",
      "45000\n",
      "50000\n",
      "55000\n",
      "60000\n",
      "65000\n",
      "70000\n",
      "75000\n",
      "80000\n",
      "85000\n",
      "90000\n",
      "5000\n",
      "245 , Luteimonas:  1.5295197308045274%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "c:\n",
      "1498 , Treponema:  3.059039461609055%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "158 , Erysipelatoclostridium:  4.588559192413582%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "484 , Ralstonia:  6.11807892321811%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "229 , Marinifilum:  7.647598654022637%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "213 , Kosakonia:  9.177118384827164%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "5000\n",
      "175 , Microvirga:  10.706638115631693%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "7 , Allofrancisella:  12.23615784643622%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "10 , Cyanothece PCC-7424:  13.765677577240746%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "20 , Archangium:  15.295197308045275%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "12 , Gimesia:  16.8247170388498%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "128 , Planomicrobium:  18.354236769654328%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "42 , Actinoallomurus:  19.883756500458855%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "36 , Oleiphilus:  21.413276231263385%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "105 , Psychroflexus:  22.942795962067912%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "12 , Planktosalinus:  24.47231569287244%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "184 , Sulfurospirillum:  26.001835423676965%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "13 , Kytococcus:  27.531355154481492%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "20 , Imperialibacter:  29.06087488528602%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "29 , Rubrivivax:  30.59039461609055%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "15 , Litoreibacter:  32.11991434689507%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "53 , Azospira:  33.6494340776996%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "10 , Hyphobacterium:  35.17895380850413%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "88 , Brachymonas:  36.708473539308656%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "35 , Ercella:  38.237993270113186%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "16 , Elev-16S-1166:  39.76751300091771%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "51 , Pseudaminobacter:  41.29703273172224%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "15 , Lactovum:  42.82655246252677%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "14 , Pseudochelatococcus:  44.35607219333129%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "28 , Kyrpidia:  45.885591924135824%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "3 , Jejuia:  47.415111654940354%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "24 , Wenzhouxiangella:  48.94463138574488%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "12 , Desulfofrigus:  50.47415111654941%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "23 , Vitellibacter:  52.00367084735393%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "92 , Ellin6055:  53.53319057815846%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "22 , Vulcaniibacterium:  55.062710308962984%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "73 , CAG-352:  56.59223003976751%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "19 , Kozakia:  58.12174977057204%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "16 , Angustibacter:  59.65126950137657%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "44 , Cedecea:  61.1807892321811%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "14 , Negativicoccus:  62.710308962985614%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "18 , Gryllotalpicola:  64.23982869379014%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "27 , Hydrogenoanaerobacterium:  65.76934842459468%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "3 , [Eubacterium] tenue group:  67.2988681553992%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "9 , Methylomarinum:  68.82838788620373%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "2 , Soortia:  70.35790761700827%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "26 , Candidatus Riegeria:  71.88742734781279%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "230 , Truepera:  73.41694707861731%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "3 , Bavariicoccus:  74.94646680942184%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "1 , Salininema:  76.47598654022637%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "63 , Caenimonas:  78.0055062710309%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "3 , Tenggerimyces:  79.53502600183542%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "2 , Koukoulia:  81.06454573263994%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "152 , DMER64:  82.59406546344448%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "16 , Candidatus Jettenia:  84.123585194249%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "48 , Candidatus Caldatribacterium:  85.65310492505354%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "1 , Crenalkalicoccus:  87.18262465585806%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "2 , Aerosakkonema Lao26:  88.71214438666259%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "2 , Maricurvus:  90.24166411746711%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "4 , Subsaximicrobium:  91.77118384827165%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "12 , RBG-16-58-14:  93.30070357907617%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "4 , Spirulina P7:  94.83022330988071%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "2 , Litorisediminivivens:  96.35974304068522%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "13 , Candidatus Electrothrix:  97.88926277148975%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "3 , Kiritimatiella:  99.41878250229428%-nÃ¡l jÃ¡r a beolvasÃ¡s.\n",
      "105341176\n"
     ]
    }
   ],
   "source": [
    "import Bio\n",
    "\n",
    "\n",
    "\"\"\"\n",
    "for temp in groups:\n",
    "    new_type = GroupedByGenus_reads(temp)\n",
    "    transformed_group.append(new_type)\n",
    "\"\"\"\n",
    "#-----------------\n",
    "\n",
    "#-----------------\n",
    "t=transformed_group[0].ListOfGenus[0].seq\n",
    "\n",
    "x_ =0\n",
    "\n",
    "for el in transformed_group:\n",
    "    x_+=1\n",
    "    if el.Genus == \"UNKNOWN\":\n",
    "        print(\"c:\")\n",
    "        continue\n",
    "    \n",
    "    #ez csak visszajelzÃ©s h hol jÃ¡r\n",
    "    if x_%50==0:\n",
    "        print(str(len(el.ListOfGenus))+\" , \"+el.Genus+\":  \" + str(x_/len(transformed_group)*100)+\"%-nÃ¡l jÃ¡r a beolvasÃ¡s.\")\n",
    "    \n",
    "    el.fill_reads()\n",
    "\n",
    "    \n",
    "\n",
    "_asd=0\n",
    "for s in transformed_group:\n",
    "#for s in range(0,1):\n",
    "\n",
    "    s_name=s.Genus\n",
    "    if \" \" in s_name:\n",
    "        s_name= s_name.replace(' ','_')\n",
    "\n",
    "    if \"-\" in s_name:    \n",
    "        s_name= s_name.replace('-','_')\n",
    "    \n",
    "    \n",
    "    path= \"Genus_4_Bacteria_20221117/\"+s_name+\"_reads.fasta\"\n",
    "    File_Genus = open(path, 'w')\n",
    "    \n",
    "    for r in s.reads:    \n",
    "        File_Genus.write(r)\n",
    "        _asd+=1\n",
    "    File_Genus.close()\n",
    "    del s\n",
    "print(str(_asd))\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
